benchmark/bench_sax.cpp

```diff
@@ -22,42 +22,197 @@ using std::cerr;
 using std::endl;
 
 const char *TWITTER_JSON = SIMDJSON_BENCHMARK_DATA_DIR "twitter.json";
-const int REPETITIONS = 10;
 
 #if SIMDJSON_IMPLEMENTATION_HASWELL
 
-#include "twitter/sax_tweet_reader.h"
+#include "twitter/tweet.h"
+#include <vector>
 
-static void sax_tweets(State &state) {
+SIMDJSON_TARGET_HASWELL
+
+namespace ondemand_bench {
+
+using namespace simdjson;
+using namespace haswell;
+
+simdjson_really_inline uint64_t nullable_int(ondemand::value && value) {
+  if (value.is_null()) { return 0; }
+  return std::move(value);
+}
+
+simdjson_really_inline twitter::twitter_user read_user(ondemand::object && u) {
+  return { u["id"], u["screen_name"] };
+}
+simdjson_really_inline void read_tweets(ondemand::parser &parser, padded_string &json, std::vector<twitter::tweet> &tweets) {
+  // Walk the document, parsing the tweets as we go
+  auto doc = parser.parse(json);
+  for (ondemand::object tweet : doc["statuses"]) {
+    tweets.emplace_back(twitter::tweet{
+      tweet["created_at"],
+      tweet["id"],
+      tweet["text"],
+      nullable_int(tweet["in_reply_to_status_id"]),
+      read_user(tweet["user"]),
```

Comment:
Can we inline nullable_int and read_user? It feels like I'd want to write.
```
tweet["in_reply_to_status_id"].is_null() ? 0 : tweet["in_reply_to_status_id"]
```

Reply:
Not with the API we have: you can only call tweet["key"] once per key--it'll keep going otherwise.

Reply:
However, I think it's so common to do this null thing that having a `get_nullable_int64(uint64_t default=0)` would be a good thing.

Reply:
To be clearer: object["key"] *always* moves forward at least one field, because the iterator has already passed the key when we handed you the value. If object["key"] was an array or object, for example, and you iterated it, we want to continue from that point most of the time.

Reply:
> However, I think it's so common to do this null thing that having a get_nullable_int64(uint64_t default=0) would be a good thing.

Or maybe we should have a coerce or "force cast" function? Kind of like "get", but one that never fails.

Reply:
> To be clearer: object["key"] always moves forward at least one field, because the iterator has already passed the key when we handed you the value. If object["key"] was an array or object, for example, and you iterated it, we want to continue from that point most of the time.

Ok. So if you have...

```
{"user":1, "user":2, "user":3}
```

Then you could do

```
mystruct.emplace_back(tweet["user"], tweet["user"], tweet["user"]);
```

to get `{1,2,3}` added. Right?

So key addressing is strictly order-dependent... if I have...

```
{"dog":1, "cat":2, "mice":3}
```
I cannot do...

```
mystruct.emplace_back(tweet["mice"], tweet["cat"], tweet["dog"]);
```

(This is less of a question and more of a "hmmm... we need to document this".)

Reply:
The "order-sensitive approach" is quite fine in a high-performance processing.

Reply:
Yes! But ultimately, while [] is really nice, it probably is the wrong syntax for it. And I do think we are going to want to support non-order-sensitive lookup, but it might be an entirely different interface.